src/m3ninx/index/segment/fst/writer.go

```diff
@@ -326,16 +393,27 @@ func (w *writer) WriteFSTTerms(iow io.Writer) error {
 	}
 
 	// make sure we consumed all the postings offsets
-	if len(offsets) != 0 {
-		return fmt.Errorf("postings offsets remain at end of terms: remaining=%d",
-			len(offsets))
+	if len(termOffsets) != 0 {
+		return fmt.Errorf("term postings offsets remain at end of terms: remaining=%d",
+			len(termOffsets))
+	}
+
+	// make sure we consumed all the postings offsets
+	if len(fieldOffsets) != 0 {
+		return fmt.Errorf("field postings offsets remain at end of terms: remaining=%d",
+			len(fieldOffsets))
 	}
 
 	// all good!
 	w.fstTermsFileWritten = true
 	return nil
 }
 
+func fieldsMetadata(fieldPostingsOffset uint64) ([]byte, error) {
+	d := fswriter.FieldData{FieldPostingsListOffset: fieldPostingsOffset}
+	return d.Marshal()
```

Comment:
Hm this is going to allocate each time yeah?

Can you create a `proto.Buffer` and a single `*fstwriter.FieldData` on the `*writer` then do:

```go
w.fieldDataProto.Reset()
w.fieldDataProto.FieldPostingsListOffset = fieldPostingsOffset
buffer.Reset()
err := buffer.Marshal(w.fieldDataProto)
```

Docs here:
https://godoc.org/github.com/golang/protobuf/proto#Buffer.Marshal

Reply:
Sure thing.

Reply:
done